### PR TITLE
Feature/refactor shop + Update class Diagram

### PR DESCRIPTION
**Pull request by Solero93**
- Simplified Shop Logic (now it only sells Units, rests of boughts are managed by class)

- Modified class Diagram accordingly

- Added Tower tag to the part of the prefab that needs it

New class Diagram:

classdiagram

### CODE REVIEW

File: Assets/Scripts/Model/Players/EasyAI.cs [hunk 1]
```diff
@@ -8,21 +8,21 @@ public class EasyAI : AI
 	public override void Play(){
         InvokeRepeating("autoCoins", 0.0f, 1.0f);
         //create 20 units every wave
-	    Shop shop = GameObject.FindGameObjectWithTag("Shop").GetComponent<Shop>();
+	    UnitShop shop = GameObject.FindGameObjectWithTag("Shop").GetComponent<UnitShop>();
         if (unitsWave < 20)
```
Comment (kkaren) [inline Assets/Scripts/Model/Players/EasyAI.cs:12]: Seems all ok! Just to know, event if it's not a change on this PR, what is unitsWave? (I don't see where it comes from)

**Review by kkaren — APPROVED** [review]
Perfect!

File: Assets/Scripts/Model/Players/EasyAI.cs (hunk 1 above)
Comment (Solero93) [reply Assets/Scripts/Model/Players/EasyAI.cs:12]: This was @JoanBenito 's thing, to say that a wave finishes after the AI spawned 20 units. It's hardcoded, shouldn't be like this. I guess @JoanBenito 'll fix this

Comment (kkaren) [reply Assets/Scripts/Model/Players/EasyAI.cs:12]: Oh ok, thanks! 😄

**Pull request merged**